content/blog/2022-06-07-using-gcp-remotes-in-dvc.md

```diff
@@ -0,0 +1,158 @@
+tags:
+  - MLOps
+  - DVC
+  - Git
+  - DVC Remotes
+  - Collaboration
+---
+
+When you’re working on a data science project that has huge datasets, it’s
+common to store them in cloud storage. You’ll also be working with different
+versions of the same datasets to train a model, so it’s crucial to have a tool
+that enables you to do this quickly and easily. That’s why we’re going to do a
+quick walkthrough of how to set up a remote in a GCP storage bucket and handle
+data versioning with [DVC](https://dvc.org/doc).
+
+We’ll start by creating a new storage bucket in our GCP account, then we’ll show
+how you can add DVC to your project, and finally, we’ll make updates to the
+dataset with DVC commands. We’ll be working with
+[this repo](https://github.com/iterative/stale-model-example) if you want an
+example to play with. By the time you finish, you should be able to create this
+setup for any ML project using an GCP remote.
+
+## Set up a GCP storage bucket
+
+Make sure that you already have a
+[GCP account](https://console.cloud.google.com). You’ll need a valid credit card
+to create a new account. Once you’re logged into your account, you should see a
+screen like this with some of the services GCP offers.
+
+![gcp_project.png](/uploads/images/2022-06-07/gcp_project.png)
+
+From here, you need to go to `Cloud Storage` to create a bucket to store the
+data. When you get to the Cloud Storage page, you should see something similar
+to this and you’ll click the `Create Bucket` button.
+
+![create_gcp_bucket.png](/uploads/images/2022-06-07/create_gcp_bucket.png)
+
+The Bucket page will have a lot of configurations you can set, but you can leave
+the settings in the default state if there’s nothing you need to customize. We
+have named this example bucket `updatedbikedata` as you can see below.
+
+![gcp_bucket_options.png](/uploads/images/2022-06-07/gcp_bucket_options.png)
+
+Now you can save your changes and you’ll be redirected to the `Bucket Details`
+page and you’ll see the bucket you just created.
+
+![created_gcp_bucket.png](/uploads/images/2022-06-07/created_gcp_bucket.png)
+
+### Get your credentials
+
+Since you have the bucket created, we need to get the credentials to connect the
+GCP remote to the project. Go to the `IAM & Admin` service and go to
+`Service Accounts` in the left sidebar. You’ll see your project and you’ll be
+able to click on `Actions` and go to where you `Manage keys` for your project.
+
+![gcp_service_account.png](/uploads/images/2022-06-07/gcp_service_account.png)
+
+Once you’ve been redirected, click the `Add Key` button and this will bring up
+the credentials you need to authenticate your GCP account with your project. Go
+ahead and download the credentials file and store it somewhere safe.
+
+![gcp_key.png](/uploads/images/2022-06-07/gcp_key.png)
+
+That’s it for setting up your storage bucket and getting the credentials you
+need! Now let’s add DVC to our demo repo and set up the remote.
```

Comment:
Are we missing a step to give the service account access to the bucket we created?

Reply:
Nope. The service account gives access to everything in this project.

Reply:
Hm, I get an error when following along:

```console
$ dvc push
ERROR: unexpected error - Forbidden: b/bikedata2/o
bicycle-service-account@extended-optics-352600.iam.gserviceaccount.com does not have storage.objects.list access to the Google Cloud Storage bucket.
```

Reply:
I had to go through a few steps to go to the bucket permissions and add the service account email with the bucket owner role to get this working.

<img width="1440" alt="Screen Shot 2022-06-06 at 9 12 19 PM" src="https://user-images.githubusercontent.com/2308172/172274820-8b8247fe-62d2-4474-9ccf-f3d2b6694530.png">
.

Reply:
That's interesting! It didn't take me through that process or throw that error. 🤔 My GCP console doesn't even have the Storage Legacy Bucket Owner role.

<img width="1737" alt="Screen Shot 2022-06-07 at 09 56 58" src="https://user-images.githubusercontent.com/47196133/172412771-29844a5a-1456-4ead-8173-b0bb18b1c12a.png">

Reply:
Yeah, there might be a more appropriate role to assign, but it seems odd that you didn't have to assign any role. Comparing to https://github.com/iterative/dvc.org/pull/3607, for example, there's a whole section there for setting the right roles on the storage account and container.

Reply:
Could simply replace "Click `Done`" with "Click `Create and Continue`" and add an Owner role specific to the newly created bucket

![image](https://user-images.githubusercontent.com/36171005/173408599-d5c667cf-1a5c-4f1c-9ebc-48fb311eda39.png)

![image](https://user-images.githubusercontent.com/36171005/173408849-3a6462db-3a05-44a1-9e4d-8bafb97db68f.png)